lib/apollo-federation/entities_field.rb

```diff
@@ -27,10 +27,9 @@ def define_entities_field(possible_entities)
     end
 
     def _entities(representations:)
-      grouped_references = representations.group_by { |r| r[:__typename] }
+      chunked_references = representations.chunk { |r| r[:__typename] }
```

Comment:
Does this change not mean that a list of representations with types alternating back and forth would mean that `resolve_references` would essentially be called _n_ times instead of two times like before?

Reply:
Yes, exactly!

Reply:
Gotcha. I wonder if there is a way we can do things like we did before, but reinstate the correct order before calling `context.schema.after_lazy` with each result. It wouldn't necessarily matter if `resolve_references` returns a lazy result, but for non-lazy implementations, this could have a decent negative impact on overall performance for large lists of representations, if the ordering is not pre-optimized by the graph router.